students/Thai_H/lesson4/mail_room_part2.py

```diff
@@ -0,0 +1,172 @@
+def thank_you_menu_selection():
+    user_selection = input('\n\nTHANK-YOU MENU\nChoose from one of options below:\n\
+        L) List current donors\n\
+        T) Send Thank-you letter (or Add donor and Send one)\n\
+        F) Create Thank-you files\n\
+        M) Back to MAIN MENU\n\
+        Q) Quit the program\n\
+        Please type L, N, or B: ')
+    return user_selection
+
+
+#-------------------------------------------------------------------------
+def thank_you_letter():
+    while True:
+        # use of dictionary where 'key' is the menu choice and 'value' is function to call
+        thank_you_menu_function = {
+        'L': create_report,
+        'T': send_thank_you_add_new_donation,
+        'F': create_thank_you_files,
+        'M': main,
+        'Q': quit_program
+    }
+        thank_you_selection = thank_you_menu_selection().upper()
+        thank_you_menu_function.get(thank_you_selection, invalid_thank_you_menu)()
+        #invalid_menu is to handle non-of-above choices
+                                                                #
+
+#--------------------------------------------------------------------------
+def sort_donor_list(unsorted_list):
+    sorted_donation_list = []
+    for d in unsorted_list:
+        total_donation =  float(sum(d['donation']))
+        number_of_donation = int(len(d['donation']))
+        average = total_donation / number_of_donation
+        sorted_donation_list.append([d['name'], total_donation, number_of_donation, average])
+    #
+    sorted_donation_list.sort(key = sort_by_total_donation, reverse = True)
+
+    return sorted_donation_list
+
+
+#--------------------------------------------------------------------------
+def sort_by_total_donation(sorted_donation_list):
+    return sorted_donation_list[1]
+
+
+#------------------------------------------------------------------------------
+def create_report():
+    sorted_list = sort_donor_list(list_of_donors)
+    report_header = ['Donor Name', 'Total Donation Amt', 'Num of Donation',  'Avg Per Donation']
+    print('\n\n{:<25}{:>20}{:>20}{:>20}'.format(*report_header))
+    print('-'*90)
+    for donor in sorted_list:
+        print('{:<26}${:>15,.2f}{:>15}{:>25,.2f}'.format(donor[0], donor[1], donor[2], donor[1] / donor[2]))
+    print('\n\n')
+
+
+#------------------------------------------------------------------------------
+def init_donor_list():
+    # create initial list of donors and their donations
+    global list_of_donors
+    list_of_donors = [
+                {'name': 'William Boeing', 'donation': [12.86, 2.00, 3.00, 4.00, 5.00]},
+                {'name': 'Steve Jobs', 'donation': [6.00, 7.00, 8.00]},
+                {'name': 'Paul Allen', 'donation': [9.00, 10.00, 11.00]},
+                {'name': 'Charles Flint', 'donation': [12.00, 13.00, 14.00]},
+                {'name': 'Thomas Edison', 'donation': [15.00, 16.00, 17.00]}
+            ]
```

Comment:
This is a list of dictionaries - for this portion of the assignment, you'll want one dictionary with names as the keys and donations as the values.